### DISCUSSION TITLE
Extending the functionality of URLCheck beyond URLs (handle more types of intents)

### DISCUSSION THREAD

**Discussion by pressRtowin** [TrianguloY/URLCheck#188] [Ideas]
Creating this discussion as @TrianguloY requested to talk more about this idea. Here's my comment from the other discussion:

> I was wondering something, specifically due to this app having URL in its name. Despite its primary functionally (checking URLs), the extensions that have been added in order to perform that function better have also pushed it into being quite a formidable handler for "Open with" and "Share" actions, to the point where I typically default to this app now, rather than others like OpenLinkWith.
> 
> That being said, there is one shortcoming in this use case, which is outside of the scope of this app's main function of checking URLs. There is no ability to handle other types of intents for non text/URL items, such as images for example, and redirect them to your preferred app. It's for this reason I still keep Better Open With around.
> 
> Would implementing features similar to that of Better Open With make sense for this URLCheck? Despite it not falling under the scope of its name, doing so would make it one of the most powerful tools out there in the broader scope of handling all Open/Share intents. You could even go a step further (though maybe as optional modules for power users) and allow for more fine control over intents such as App Manager does as I mentioned here.

And TrianguloY's thoughts on the matter:

> That's a very long term request, and even though I'm not sure if my free time will allow me, it looks like a good idea. I made the app modular to quickly add new features, and so far it's been helping with it. Unfortunately you need to draw the line somewhere, and I drew it at url manager. Doesn't mean allowing more intents isn't possible, just that's it'll be a bit harder.

Even if this is handled over a long period of time, I think it's a fairly exciting idea. URLCheck has already allowed me to remove several other apps from my main work flow and day-to-day phone usage that all did similar but slightly different things. Handling other type of intents is the last big thing not covered.

**Comment by pressRtowin** [comment]
I had a thought about this while thinking about some of the new issues being discussed. If this idea is something you'd be interested in exploring in the future, is there any value in coding things in such a way where it isn't assumed that the intent being passed on is a URL? I'd assume most of the code now operates on the assumption that some kind of URL was passed to URLCheck, either through the Open or Share actions. If other intent types were preemptively allowed for where applicable, while still limiting actual functionality to URLs, could that make implementing this idea in the future any easier, or would it be too much of a change or too much work to do so now?

Hopefully I explained that in a way that makes sense.

**Reply by TrianguloY** [reply]
Currently the app accepts any string as input, you can test this by sharing any text or simply writing in the input module. Still, most modules expect a valid url and they usually report errors on other intents.
Most actions are also configured to trigger with urls, like being a default browser instead of a default image viewer. 

In any case this is a big change that, as I said, most probably won't happen in the near future. In any case the app is open source so anyone is encouraged to improve it or even create another one with these ideas.
